### PR TITLE
[#91] feat: weekly stats calculations integration test

### LINKED ISSUE #91: [Epic 8] Tests for GitHub weekly stats computation
## Description

Write tests for lib/github-weekly-stats.ts covering the core behaviour and edge cases.

## Acceptance Criteria

- [ ] Commit and PR counts are correctly aggregated per project
- [ ] Lines added/removed are summed correctly across all commits
- [ ] Contributors are correctly mapped from identities through to project members
- [ ] Projects with no repos are skipped without error
- [ ] Weekly stats and member contributions are upserted without error

## Dependencies/Blockers

#87

### PR DESCRIPTION
**Pull request by NoobCrewDelux**
## Description

Runs an integration test with various cases against github-weekly-stats.ts

## Notes

Does not address edge case where there maybe me multiple repos per project. At the time of posting I'm not sure if it's a probable that this would ever happen therefore I decided to leave it.

### CODE REVIEW

File: apps/worker/src/lib/github-weekly-stats.integration.test.ts [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+  })
+
+  it('aggregates commit and PR counts correctly for a single project', async () => {
+    const { project, repo } = await seedProjectWithRepo()
+    const { person, identity } = await seedPersonWithIdentity('dev1')
+    await seedProjectMember(project.id, person.id)
+
+    await seedCommitFact(repo.id, identity.id, { sha: 'c1' })
+    await seedCommitFact(repo.id, identity.id, { sha: 'c2' })
+    await seedCommitFact(repo.id, identity.id, { sha: 'c3' })
```
Comment (OshanpNZ) [inline apps/worker/src/lib/github-weekly-stats.integration.test.ts]: We exclude main commits on purpose, but nothing here checks that.  Can we seed a couple of commits on main/master + a null branch and assert they don't get counted? Helper already takes a branch arg.

File: apps/worker/src/lib/github-weekly-stats.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+    expect(contribA!.linesRemoved).toBe(10)
+
+    expect(contribB).not.toBeNull()
+    expect(contribB!.commits).toBe(1)
+    expect(contribB!.prsMerged).toBe(0)
+    expect(contribB!.linesAdded).toBe(20)
+    expect(contribB!.linesRemoved).toBe(2)
+  })
+
+  it('does not create a contribution row for an identity with no matching project member', async () => {
```
Comment (OshanpNZ) [inline apps/worker/src/lib/github-weekly-stats.integration.test.ts:179]: Could we also cover a commit with authorIdentityId = null? The helper supports passing null. The expected behaviour is that it still counts toward WeeklyStats.commits/linesAdded but produces no member contribution row

File: apps/worker/src/lib/github-weekly-stats.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+    await computeWeeklyGitHubMetrics(input, WEEK_START, WEEK_END)
+    await expect(computeWeeklyGitHubMetrics(input, WEEK_START, WEEK_END)).resolves.not.toThrow()
+
+    const contribs = await db.memberWeeklyContribution.findMany({
+      where: { projectMemberId: member.id, weekStart: WEEK_START },
+    })
+    expect(contribs).toHaveLength(1)
+  })
+
+  it('selects the member with the most lines changed as MVP', async () => {
```
Comment (OshanpNZ) [inline apps/worker/src/lib/github-weekly-stats.integration.test.ts]: The MVP selection has a 3-level tie-break (lines changed → commit count → lowest projectMemberId), but only the straightforward "most lines" path is tested. Could we add at least one test where two members tie on lines changed but differ on commits, to lock down the tie-break?

**Review by OshanpNZ — CHANGES_REQUESTED** [review]

File: apps/worker/src/lib/github-weekly-stats.integration.test.ts [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+    })
+
+    const contribs = await db.memberWeeklyContribution.findMany({
+      where: { weekStart: WEEK_START },
+    })
+
+    expect(contribs).toHaveLength(0)
+
+    if (!stats) throw new Error('Expected weekly stats to exist')
+    expect(stats.linesAdded).toBe(20)
```
Comment (kelvinchen8) [inline apps/worker/src/lib/github-weekly-stats.integration.test.ts:204]: `expect(stats.commits).toBe(2)` should also be here right?

**Review by kelvinchen8 — CHANGES_REQUESTED** [review]

**Review by OshanpNZ — APPROVED** [review]
lgtm

**Review by kelvinchen8 — APPROVED** [review]
lgtm
